apollo-router/src/plugins/telemetry/tracing/datadog.rs

```diff
@@ -0,0 +1,76 @@
+use crate::plugins::telemetry::config::Trace;
+use crate::plugins::telemetry::{GenericWith, TracingConfigurator};
+use opentelemetry::sdk::trace::Builder;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use tower::BoxError;
+use url::Url;
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    pub endpoint: AgentEndpoint,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", untagged)]
+pub enum AgentEndpoint {
+    Default(AgentDefault),
+    Url(Url),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+pub enum AgentDefault {
+    Default,
+}
```

Comment:
these enums are common between Jaeger, Zipkin and Datadog, maybe we could keep only one version and use it everywhere